### PR TITLE
feat: allow specification of module style (esm / cjs)

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Currently the generator uses the CommonJS module format. With the ECMAScript Module format becoming more widely supported it would be good to support both.

This PR generates the module format based on an environment variable. The goal is to supply these values when running the `forge` command.

### CODE REVIEW

File: helpers/setVarBlock.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+function setVar(varName, options) {
```
Comment (ms14981) [inline helpers/setVarBlock.js:1]: I'm not quite sure what the purpose of this is? For the modelIncludes file, I would assume that we're trying to achieve

commonjs:
```
const {
  // ...
} = require("...");
```

es6:
```
import {
  // ...
} from "...";
```
What am I missing here?

File: helpers/import.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+function importHelper(type, file) {
+  switch (process.env.moduleFormat) {
+    case "esm":
+      return `import ${type} from "./${file}.js"`;
```
Comment (ms14981) [inline helpers/import.js]:
```suggestion
      return `import ${type} from "./${file}"`;
```

File: helpers/import.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+function importHelper(type, file) {
+  switch (process.env.moduleFormat) {
```
Comment (ms14981) [inline helpers/import.js:2]: Can we add a line to the JS README mentioning this option (and maybe an example of it in use)?

File: helpers/import.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    case "esm":
+      return `import ${type} from "./${file}.js"`;
+    case "cjs":
```
Comment (ms14981) [inline helpers/import.js]: Personally I think I'd prefer longer option names, e.g. "esmodule" or "commonjs", but I'm not too fussed :)

File: helpers/import.js (hunk 3 above)
Comment (ColinEberhardt) [reply helpers/import.js:2]: Good point - I've added information to the README about how to list the generator options. In this case I think the effect should be relatively obvious, so have not included an example.

File: helpers/import.js (hunk 4 above)
Comment (ColinEberhardt) [reply helpers/import.js]: I think that's a fair point. I've updated with the long names

File: helpers/setVarBlock.js (hunk 1 above)
Comment (ColinEberhardt) [reply helpers/setVarBlock.js:1]: Yes, that is what we're looking to achieve. The `import` handlebars helper generates the boilerplate you detailed. However, you need to pass in the names of the objects to import. If the name is something that is constructed dynamically by the template we need a way to capture that.

**Review by ms14981 — APPROVED** [review]

**Pull request merged**